tests/components/mqtt/test_discovery.py

```diff
@@ -481,3 +442,75 @@ async def test_complex_discovery_topic_prefix(hass, mqtt_mock, caplog):
     assert state is not None
     assert state.name == "Beer"
     assert ("binary_sensor", "node1 object1") in hass.data[ALREADY_DISCOVERED]
+
+
+async def test_mqtt_integration_discovery_subscribe_unsubscribe(
+    hass, mqtt_client_mock, mqtt_mock
+):
+    """Check MQTT integration discovery subscribe and unsubscribe."""
+    mock_entity_platform(hass, "config_flow.comp", None)
+
+    entry = hass.config_entries.async_entries("mqtt")[0]
+    mqtt_mock().connected = True
+
+    with patch(
+        "homeassistant.components.mqtt.discovery.async_get_mqtt",
+        return_value={"comp": ["comp/discovery/#"]},
+    ):
+        await async_start(hass, "homeassistant", entry)
+        await hass.async_block_till_done()
+
+    mqtt_client_mock.subscribe.assert_any_call("comp/discovery/#", 0)
+    assert not mqtt_client_mock.unsubscribe.called
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6299130)